src/transformers/modeling_deberta.py

```diff
@@ -0,0 +1,1061 @@
+        drop_prob (float): the dropout probabilities
+
+    """
+
+    def __init__(self, drop_prob):
+        super().__init__()
+        self.drop_prob = drop_prob
+        self.count = 0
+        self.context_stack = None
+
+    def forward(self, x):
+        """Call the module
+
+        Args:
+            x (:obj:`torch.tensor`): The input tensor to apply dropout
+
+
+        """
+        if self.training and self.drop_prob > 0:
+            return XDropout.apply(x, self.get_context())
+        return x
+
+    def clear_context(self):
+        self.count = 0
+        self.context_stack = None
+
+    def init_context(self, reuse_mask=True, scale=1):
+        if self.context_stack is None:
+            self.context_stack = []
+        self.count = 0
+        for c in self.context_stack:
+            c.reuse_mask = reuse_mask
+            c.scale = scale
+
+    def get_context(self):
+        if self.context_stack is not None:
+            if self.count >= len(self.context_stack):
+                self.context_stack.append(DropoutContext())
+            ctx = self.context_stack[self.count]
+            ctx.dropout = self.drop_prob
+            self.count += 1
+            return ctx
+        else:
+            return self.drop_prob
+
+
+def MaskedLayerNorm(layerNorm, input, mask=None):
+    """Masked LayerNorm which will apply mask over the output of LayerNorm to avoid inaccurate updatings to the LayerNorm module.
+    Args:
+      layernorm (:obj:`~DeBERTa.deberta.BertLayerNorm`): LayerNorm module or function
+      input (:obj:`torch.tensor`): The input tensor
+      mask (:obj:`torch.IntTensor`): The mask to applied on the output of LayerNorm where `0` indicate the output of that element will be ignored, i.e. set to `0`
+
+    Example::
+
+      # Create a tensor b x n x d
+      x = torch.randn([1,10,100])
+      m = torch.tensor([[1,1,1,0,0,0,0,0,0,0]], dtype=torch.int)
+      LayerNorm = DeBERTa.deberta.BertLayerNorm(100)
+      y = MaskedLayerNorm(LayerNorm, x, m)
+
+    """
+    output = layerNorm(input).to(input)
+    if mask is None:
+        return output
+    if mask.dim() != input.dim():
+        if mask.dim() == 4:
+            mask = mask.squeeze(1).squeeze(1)
+        mask = mask.unsqueeze(2)
+    mask = mask.to(output.dtype)
+    return output * mask
+
+
+class BertLayerNorm(nn.Module):
+    """LayerNorm module in the TF style (epsilon inside the square root)."""
+
+    def __init__(self, size, eps=1e-12):
+        super().__init__()
+        self.weight = nn.Parameter(torch.ones(size))
+        self.bias = nn.Parameter(torch.zeros(size))
+        self.variance_epsilon = eps
+
+    def forward(self, x):
+        input_type = x.dtype
+        x = x.float()
+        u = x.mean(-1, keepdim=True)
+        s = (x - u).pow(2).mean(-1, keepdim=True)
+        x = (x - u) / torch.sqrt(s + self.variance_epsilon)
+        x = x.to(input_type)
+        y = self.weight * x + self.bias
+        return y
+
+
+class BertSelfOutput(nn.Module):
```

Comment:
Can we call all class names, such as `BertSelfOutput` -> `DeBertSelfOutput` and `BertLayerNorm` -> `DeBertLayerNorm`?

Reply:
I'm OK with that.

Reply:
I would prefer `Deberta` as a name prefix.

Reply:
Also, if all of those are identical copies from the bert modeling file, they should have the proper comment to enable the copy check script.